### PR TITLE
Spark: Clarify schema behavior when working with branches

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: docs/docs/spark-queries.md [hunk 1]
```diff
@@ -124,6 +124,27 @@ SELECT * FROM prod.db.table.`tag_historical-snapshot`;
...
 Note that the identifier with branch or tag may not be used in combination with `VERSION AS OF`.
 
+
+#### Schema selection in time travel queries
+
+The different time travel queries mentioned in the previous section can use either the snapshot's schema or the table's schema
+
+```sql
+-- time travel to October 26, 1986 at 01:21:00 -> uses the snapshot's schema
+SELECT * FROM prod.db.table TIMESTAMP AS OF '1986-10-26 01:21:00';
```
Comment (manuzhang) [inline docs/docs/spark-queries.md:134]: Can we have some example outputs to show the difference if any?

File: docs/docs/spark-writes.md [hunk 2]
```diff
@@ -201,6 +201,10 @@ Note WAP branch and branch identifier cannot both be specified.
 Also, the branch must exist before performing the write. 
 The operation does **not** create the branch if it does not exist. 
 For more information on branches please refer to [branches](branching.md).
+
+!!! info
```
Comment (manuzhang) [inline docs/docs/spark-writes.md:205]: I think we also need some clarifications at https://iceberg.apache.org/docs/nightly/branching/

File: docs/docs/spark-queries.md (hunk 1 above)
Comment (nastra) [reply docs/docs/spark-queries.md:134]: I've added a full example to the branching page

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/TestSelect.java [hunk 3]
```diff
@@ -348,6 +349,64 @@ public void testBranchReference() {
     assertEquals("Snapshot at specific branch reference name", expected, fromDF);
   }
 
+  @TestTemplate
+  public void readAndWriteWithBranchAfterSchemaChange() {
```
Comment (rdblue) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/TestSelect.java:353]: This test looks correct to me. Thanks, @nastra!

**Review by rdblue — APPROVED** [review]

**Comment by nastra** [comment]
thanks for the reviews @manuzhang and @rdblue

**Pull request merged**